clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -109,6 +109,9 @@ invariant_checker_t::parallel_shard_init(int shard_index, void *worker_data)
 bool
 invariant_checker_t::parallel_shard_exit(void *shard_data)
 {
+    per_shard_t *shard = reinterpret_cast<per_shard_t *>(shard_data);
```

Comment:
As discussed offline, a better fix would be to use instr_autoclean_t (similar to instrlist_autoclean_t used elsewhere in our codebase) so that unique_ptr itself can do this cleanup.

Reply:
Done. Added `instr_autoclean_t` struct. not sure if we should move it outside the invariant checker and make it available to be used elsewhere.